src/plugins/engagement/public/plugin.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import type { CoreStart, Plugin, PluginInitializerContext } from 'src/core/public';
+import type {
+  EngagementPluginSetup,
+  EngagementPluginStart,
+  EngagementPluginSetupDeps,
+  EngagementPluginStartDeps,
+} from './types';
+import type { EngagementServices } from './services';
+import { ServicesProvider } from './services';
+import type { GetChatTokenResponseBody } from '../common';
+import { GET_CHAT_TOKEN_ROUTE_PATH } from '../common';
+
+export interface EngagementConfigType {
+  chat: {
+    enabled: boolean;
+    chatURL: string;
+  };
+}
+
+interface ChatUser {
+  id: string;
+  email: string;
+}
+
+export class EngagementPlugin
+  implements
+    Plugin<
+      EngagementPluginSetup,
+      Promise<EngagementPluginStart>,
+      EngagementPluginSetupDeps,
+      EngagementPluginStartDeps
+    >
+{
+  constructor(private readonly initializerContext: PluginInitializerContext) {}
+
+  public setup(): EngagementPluginSetup {
+    return {};
+  }
+
+  private async getChatUser(): Promise<ChatUser | null> {
+    // TODO: obtain kibana and/or cloud user information
+    return {
+      id: 'user-id',
+      email: 'test-user@elasticsearch.com',
+    }
+  }
```

Comment:
We need to find a way to obtain a Kibana or Cloud user data here. Particularly we need to get user ID (might be Kibana user's `username`) and user email (this should be the same as in Cloud for Cloud users)
@pgayvallet maybe you know how to do this?